### PR TITLE
chore(deps): Update to Rust 1.58

### PR DESCRIPTION
**Pull request by jszwedko**
Mostly `clippy --fix`

Closes: #10852

Signed-off-by: Jesse Szwedko <jesse@szwedko.me>

### CODE REVIEW

**Comment by jszwedko** [comment]
> Did you grep for everywhere we may be using 1.57 in dockerfiles and ci config?

👍 I did. You can see those updates here. I think I got them all.

**Comment by spencergilbert** [comment]
> > Did you grep for everywhere we may be using 1.57 in dockerfiles and ci config?
> 
> 👍 I did. You can see those updates here. I think I got them all.

I had a vague recollection about having the version in different places when Jeremie was doing the docker-compose work, I just couldn't remember where all we identified :P

File: lib/vector-core/buffers/src/disk_v2/reader.rs [hunk 1]
```diff
@@ -243,9 +243,7 @@ where
     /// to an out-of-order read.
     pub fn read_record(&mut self, token: ReadToken) -> Result<T, ReaderError<T>> {
         let record_id = token.into_id();
-        if record_id != self.current_record_id {
-            panic!("using expired read token");
-        }
+        assert!(!(record_id != self.current_record_id), "using expired read token");
```
Comment (bruceg) [inline lib/vector-core/buffers/src/disk_v2/reader.rs:246]: That's a smart clippy lint.

**Review by bruceg — APPROVED** [review]

**Comment by jszwedko** [comment]
> > > Did you grep for everywhere we may be using 1.57 in dockerfiles and ci config?
> > 
> > 
> > 👍 I did. You can see those updates here. I think I got them all.
> 
> I had a vague recollection about having the version in different places when Jeremie was doing the docker-compose work, I just couldn't remember where all we identified :P

He smartly was able to tie it back to `RUST_VERSION` set in the `Makefile` by extracting it from `rust-toolchain`.

**Review by blt — APPROVED** [review]

**Pull request merged**